modules/ugdk-core/src/action/scene.cc

```diff
@@ -45,14 +48,31 @@ void Scene::Focus() {
 void Scene::DeFocus() {
     event_handler_.RaiseEvent(SceneDefocusEvent(this));
 }
-
 void Scene::Finish() {
     if (finished_)
         return;
     End();
     event_handler_.RaiseEvent(SceneFinishedEvent(this));
     finished_ = true;
 }
+uint32_t Scene::num_functions() {
+    return render_functions_.size();
+}
+void Scene::set_render_function(uint32_t index,
+                                const std::function<void (graphic::Canvas&)>& function) {
+    assert(index <= render_functions_.size()); // check if it's not too big, if it's in the edge then it
+                                               // push_backs
+    if (index == render_functions_.size())
+        render_functions_.emplace_back(function);
+    else
+        render_functions_[index] = function;
+    /* Check if there are enough render-targets ? */
+}
+void Scene::remove_render_function(uint32_t index) {
```

Comment:
This is neither a getter nor a setter, so the name formatting should be PascalCase.

Reply:
We have a little problem with the example now